### PR TITLE
Expand method with iteration count, as mentioned in #18.

### PR DESCRIPTION
**Pull request by ShinJJang**
Hello there, 

As mentioned in #18, default constructor of SecurePreference is too slow to create instance. In there, solution is to give parameter at `AesCbcWithIntegrity.generateKeyFromPassword(...)` like below.

```
AesCbcWithIntegrity.SecretKeys mykeys = AesCbcWithIntegrity.generateKeyFromPassword(password, salt, iterationCount);
SecurePreferences securePrefs = new SecurePreferences(getContext(), mykeys, "pref-file");
```

But if **already use this library**, we need to change password using `handlePasswordChange(String newPassword, Context context)` **to solve the problem too slow to create instance**. This method doesn't support not only iteration count to generate key, but also AES key.

If I solve above problem, next is to check is this migrated. Then if password changed with iteration, I make instance of SecurePreference. But also, there is blocker `generateAesKeyName` in constructor when pass password, not key. `generateAesKeyName` is default key generator of key, also don't support iteration count to generate key. But this method is **private static**. I can't override it. So inheritance of SecurePreference can't solve problem.

## So What I did :

1. Expand method `constructor, handlePasswordChange, generateAesKeyName` with iterationCount

2. Make public `generateAesKeyName` for to prepare something to change logic
(Maybe, protected is enough to cover this issue.)

3. Add comment

4. Add test to `handlePasswordChange` with iteration count

5. Fix wrong method name. `getUnencryptedString` -> `getEncryptedString` 
(Value from sharePreference look like not encrypted, but not using original key, using hashPrefKey in this method)

6. Fix wrong assignment to sharePrefFilename (Same with #42)

Thanks for reading. If you have any question or request, please let me know.

### CODE REVIEW

File: library/src/main/java/com/securepreferences/SecurePreferences.java [hunk 1]
```diff
@@ -192,21 +194,23 @@ public void destroyKeys(){
     /**
      * Uses device and application values to generate the pref key for the encryption key
      * @param context
-     * @return String to be used as the AESkey Pref key
+     * @param iterationCount
+	 * @return String to be used as the AESkey Pref key
      * @throws GeneralSecurityException if something goes wrong in generation
      */
-	private static String generateAesKeyName(Context context) throws GeneralSecurityException
+	public String generateAesKeyName(Context context, int iterationCount) throws GeneralSecurityException
```
Comment (lala7573) [inline library/src/main/java/com/securepreferences/SecurePreferences.java]: why did you change 'private static' keyword to 'public' ?

Comment (ShinJJang) [inline library/src/main/java/com/securepreferences/SecurePreferences.java]: @lala7573  Because I add parameter iteration count in `generateAesKeyName`, don't have to make public, as you said earlier. But, I think that logic to generate the pref key for the encryption key can editable for some case or someone's intention. Also I know `private static` is used for internal utilization not depended on business logic. So `public` is not necessary, but must be `protected` or `private` at least.
